api/src/catalog/catalog.d.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file User type declarations.
+ */
```

Comment:
Don't use `.d.ts` for type declarations—use a normal `.ts` file instead. Also where were these types before? Did they not exist?

Reply:
Done :)

We can clean up explicitly defined types after you implement the runtime lib.